patterns/creational/builder/main.go

```
@@ -0,0 +1,147 @@
+package main
+
+import "fmt"
+
+type pizza struct {
+	doughType   string
+	sauseType   string
+	sausageType string
+	cheeseType  string
+}
+
+type pizzaBuilder interface {
+	setDoughType()
+	setSauseType()
+	setSausageType()
+	setCheeseType()
+	getPizza() pizza
+}
+
+func getPizzaMaker(pizzaType string) pizzaBuilder {
+	if pizzaType == "pepperoni" {
+		return &pepperoniBuilder{}
+	}
+
+	if pizzaType == "vesuvius" {
+		return &vesuviusBuilder{}
+	}
+
+	return nil
```

i'd say we should return an error here

---

done